app/helpers/application_helper.rb

```diff
@@ -63,6 +63,17 @@ def field_or_none(label, value, tag: :p, tag_options: {}, separator: ': ',
                      label_class: label_class, value_class: value_class)
   end
 
+  def icon_link(url, icon, disabled_class = 'disabled_icon')
+    if icon.nil?
+      nil
+    elsif !url.blank?
+      link_to raw("<i class='fab fa-#{icon} fa-2x'></i>"), url, target: '_blank'
+    else
+      raw("<i class='fab fa-#{icon} fa-2x #{disabled_class}'></i>")
+    end
+  end
+  
+
```

Comment:
Revert the logic back to how you had it (before I asked you to change it to display the disabled version if url was blank)

- Instead of using `raw(...)` it's better to use an existing helper method.
  The font-awesome-sass gem has a helper method that will return the HTML for one of their icons:  `icon(...)`   I put that in place.

- I explicitly return `nil` to make it clear to future readers of this code that these conditions are addressed (`url.blank? || icon.nil?`) and that `nil` is what we intended.  It also makes it easy to change what is returned if we need to do so later.  (e.g. What if later we decided than an empty string is a better return value in that case?)

- In general, don't use `if <negated condition.`  Use `unless <positive condition>` instead  


```
    if url.blank? || icon.nil?
      nil  
    else
      link_to icon('fab', icon, {class: 'fa-2x'}), url, target: '_blank'
    end
```

We could make `'fab'` and `'fa-2x'` constants, but I'm fine just doing this.  It's quite clear to future readers, and I don't think we'd gain anything in code efficiency.